### PR TITLE
Documented the dist folder

### LINKED ISSUE #4242: sbt-native-packager doesn't copy configuration when play plugin is enabled
The sbt-native-packager docs say [here](http://www.scala-sbt.org/sbt-native-packager/archetypes/java_server/customize.html#linux-configuration) that you can create an `application.ini` in `src/universal/conf` and it will process it and put it in the app's `conf` directory. When the play plugin is enabled this file does not end up in the conf directory. The sbt guys said this seems to be a problem with play.

https://github.com/Kyprus/sbt-native-packager-test demonstrates the issue

### PR DESCRIPTION
**Pull request by jroper**
Fixes #4242

Also moved configuration of the dist folder to the PlayLayoutPlugin, so that if disabled, it will revert to the standard `src/universal`.

### CODE REVIEW

File: documentation/manual/gettingStarted/Anatomy.md [hunk 1]
```diff
@@ -140,6 +141,7 @@ src                        → Application sources
     └ java                 → Java source folder for unit or functional tests
     └ scala                → Scala source folder for unit or functional tests
     └ resources            → Resource folder for unit or functional tests
+ └ universal               → Arbitrary files to be included in your projects distribution
```
Comment (benmccann) [inline documentation/manual/gettingStarted/Anatomy.md]: What's the difference between dist and public/universal? I wouldn't know what to put where given that they have the same description

Comment (benmccann) [inline documentation/manual/gettingStarted/Anatomy.md]: Nevermind. Stupid comment. The github diff view hid from me that one line is with PlayLayoutPlugin enabled and one line is with it disabled. This makes perfect sense when I see the whole page

File: documentation/manual/detailedTopics/production/Production.md [hunk 2]
```diff
@@ -110,9 +110,13 @@ Then build your package with:
 
 > There will be some error logging. This is because rpm logs to stderr instead of stdout.
 
+### Including additional files in your distribution
+
+Anything included in your projects `dist` directory will be included in the distribution built by the native packager.  Note that in Play, the `dist` directory is equivalent to the `src/universal` directory mentioned in the native packagers own documentation.
```
Comment (benmccann) [inline documentation/manual/detailedTopics/production/Production.md]: packagers -> packager's

Comment (jroper) [inline documentation/manual/detailedTopics/production/Production.md]: Updated.

**Pull request merged**